actionpack/lib/action_dispatch/testing/integration.rb

```diff
@@ -89,40 +113,55 @@ def follow_redirect!
       # redirect. Note that the redirects are followed until the response is
       # not a redirect--this means you may run into an infinite loop if your
       # redirect loops back to itself.
-      def request_via_redirect(http_method, path, parameters = nil, headers_or_env = nil)
-        process(http_method, path, parameters, headers_or_env)
+      #
+      # Example:
+      #
+      #   request_via_redirect :post, '/welcome',
+      #     params: { ref_id: 14 },
+      #     headers: {"X-Test-Header" => "testvalue"}
+      def request_via_redirect(http_method, path, *args)
+        if kwarg_request?(*args)
```

Comment:
It looks like you are duplicating some logic from `process_with_kwargs`, is it intended ? Otherwise, the method could look like:

~~~ruby
def request_via_redirect
  process_with_kwargs(http_method, path, *args)

  follow_redirect! while redirect?
  status
end
~~~